### PR TITLE
CSCETSIN-61: DataCite mapping

### PR DESCRIPTION
**Pull request by Meepu**
Map DataCite to Metax.

Includes one test file, could have more.

See https://wiki.eduuni.fi/pages/viewpage.action?pageId=52822738 for full mapping table.

### CODE REVIEW

File: ckanext/etsin/mappers/datacite.py [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+        # save it in language field
+        language = "default"
+
+    # Map title
+    # In case of multiple primary titles, pick only the first one
+    for title in xml.findall('.//title'):
+        title_type = xml.find('.//title').get('titleType')
+        if not title_type:
+            # Title is primary when there's no titleType
+            package_dict['title'] = [{language: title.text}]
```
Comment (junsk1) [inline ckanext/etsin/mappers/datacite.py]: title's cardinality is 1, so array should be left out?

File: ckanext/etsin/mappers/datacite.py [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+
+    # Map alternate identifier
+    package_dict['other_identifier'] = []
+    for alternate_identifier in xml.findall('.//alternateIdentifier'):
+        alternate_identifier_type = alternate_identifier.get(
+            'alternateIdentifierType')
+        if alternate_identifier_type == "URL":
+            package_dict['other_identifier'].append({
+                'notation': alternate_identifier.text,
+                'type': alternate_identifierType,
```
Comment (junsk1) [inline ckanext/etsin/mappers/datacite.py]: Type is an object, which should have an identifier derived from identifier_type reference data.

File: ckanext/etsin/mappers/datacite.py [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+    if version is not None:
+        package_dict['version_info'] = version.text
+
+    # Map rights
+    package_dict['access_rights'] = []
+    for right in xml.findall('.//rights'):
+        rights_URI = right.get('rightsURI')
+        package_dict['access_rights'].append({
+            'description': right.text,
+            'license': {'identifier': rights_URI},
```
Comment (junsk1) [inline ckanext/etsin/mappers/datacite.py]: Can we be sure rights_URI is something that is found in license reference data?

**Review by junsk1 — CHANGES_REQUESTED** [review]

File: ckanext/etsin/mappers/datacite.py (hunk 1 above)
Comment (Meepu) [reply ckanext/etsin/mappers/datacite.py]: Right.

File: ckanext/etsin/mappers/datacite.py (hunk 2 above)
Comment (Meepu) [reply ckanext/etsin/mappers/datacite.py]: Identifier_type.json currently has an entry with ID "url". Should we check every time that the entry is still in reference data?

Comment (Meepu) [reply ckanext/etsin/mappers/datacite.py]: Also: I fixed the typo in variable name.

File: ckanext/etsin/mappers/datacite.py (hunk 3 above)
Comment (Meepu) [reply ckanext/etsin/mappers/datacite.py]: No, we can't. Then again, none of the ids in the license reference data are URIs, so I'm not sure if using them would be any better.

What's the policy on checking reference data? Kielipankki refiner doesn't seem to check anything, it just assumes that certain values will always be in reference data.

**Review by junsk1 — APPROVED** [review]

**Pull request merged**